io/js/src/main/scala/fs2/io/internal/SuspendedStream.scala

```diff
@@ -26,7 +26,7 @@ package internal
 import cats.effect.kernel.Concurrent
 import cats.effect.kernel.Resource
 import cats.effect.std.Queue
-import cats.effect.std.Semaphore
+import cats.effect.std.Mutex
```

Comment:
This switch from `Semaphore` to `Mutex` seems unrelated to main PR. Perhaps separate PR?